### ISSUE TITLE
FindHIP gives CUDA-error on AMD systems

### ISSUE THREAD

**Issue by VincentSC** [ROCm-Developer-Tools/HIP#137]
When cmake cannot find HIP installed, it gives:
> CMake Error at /usr/share/cmake-3.5/Modules/FindCUDA.cmake:617 (message):
  Specify CUDA_TOOLKIT_ROOT_DIR

This is not a good feedback, as it assumes the user knows that it should be ignored when no NVIDIA GPUs are installed.

A better message would be: "Both HIP and CUDA are not found. In case HIP is targeted, make sure it's installed correctly. In case CUDA is targeted, try specifying CUDA_TOOLKIT_ROOT_DIR" 

Alternatively It could check for currently installed drivers/GPUs.

**Comment by mangupta** [comment]
FindHIP does not use FindCUDA. The above error message is being issued by FindCUDA. The error is likely coming from your cmake project explicitly including find_package(CUDA).

**Comment by VincentSC** [comment]
Please don't close when you assume we use FindCUDA. We use FindHIP, and this is the error we get.

**Comment by mangupta** [comment]
Can you point to the cmake project you are using? The error you posted originates from FindCUDA. FindHIP does not include or use FindCUDA in anyway. I am reopening the issue.

**Comment by VincentSC** [comment]
Could you get in contact with your colleague Bragadeesh Natarajan? He can get you access to the cmake-file.

**Comment by mangupta** [comment]
@VincentSC I obtained the code from Bragadeesh. Running ```grep -Ri find_package``` inside the codebase returned
```
cmake/Dependencies.cmake:find_package(Git REQUIRED)
cmake/Dependencies.cmake:find_package(HIP REQUIRED)
cmake/Modules/FindHIP.cmake:find_package_handle_standard_args(
cmake/Modules/FindHIP.cmake:    find_package(CUDA QUIET REQUIRED)
```

Line 174 of your copy of FindHIP.cmake seems to include FindCUDA.cmake. Looking at the history of FindHIP.cmake in the HIP repository shows that these lines were never present. So looks like it was added explicitly by someone in your codebase. git log shows that it was added by Adel Johar.

The solution is to replace your copy of FindHIP.cmake with the one from the HIP repo. Let me know if this solves your issue.
